iocore/net/SSLSNIConfig.cc

```diff
@@ -73,7 +74,14 @@ SNIConfigParams::loadSNIConfig()
     aiVec->push_back(ai1);
     auto ai2 = new VerifyClient(item.verify_client_level);
     aiVec->push_back(ai2);
-    sni_action_map.put(ats_strdup(servername), aiVec);
+    if (wildcard) {
```

Comment:
Rather than using the wildcard matcher and this check, what about
```
static constexpr auto WILDCARD_PREFIX = "*."_sv;
if (domain.size() > WILDCARD_PREFIX.size() &&
    WILDCARD_PREFIX == domain.substr(0,WILDCARD_PREFIX.size())
  ) {
  wild_sni_action_map.put(ats_stringdup(domain.substr(WILDCARD_PREFIX.size()), aiVec);
}
```